File: JS/709-to-lower-case.js
/**
 * @param {string} str
 * @return {string}
 */
var toLowerCase = function(str) {
  return str.split("").map((x) => (x.charCodeAt(0) > 64 && x.charCodeAt(0) < 91) ? x = String.fromCharCode(x.charCodeAt(0) + 32) : x).join("")
  
};

// I code golf'd this one. The incoming string is split on each charcater in an array. 
// The array is mapped and each character's value is checked. 
// Uppercase characters are valued from 65 to 90. If the character is uppercase, add 32
// to the ASCII value to make it lowercase. 
// if the character's value is >= 92 then do nothing
// all characters are joined back into a returned string

File: PY/709-to-lower-case.py
class Solution:
    def toLowerCase(self, str: str) -> str:
        modified = [] # strings are immutable, so the characters in str will be added to this list
        for char in str: # standard for...in loop
          if ord(char) > 64 and ord(char) < 91: # uppercase ASCII is 65 to 90 for A-Z
            char = chr(ord(char) + 32) # lowercase ASCII is 32 additional values
          modified.append(char) # append the char to the list
          final = ''.join(modified) # the output is expecting a string, so join modified with no punctuation or spaces
        return(final)